api/src/services/applications/create_application.py

```diff
@@ -68,9 +69,41 @@ def create_application(db_session: db.Session, competition_id: UUID) -> Applicat
 
     # Create a new application
     application = Application(application_id=uuid.uuid4(), competition_id=competition_id)
-
     db_session.add(application)
 
+    # Associate user with application if provided
+    if user_id:
+        # Check if the user-application association already exists
+        existing_association = db_session.execute(
+            select(ApplicationUser).where(
+                ApplicationUser.application_id == application.application_id,
+                ApplicationUser.user_id == user_id,
+            )
+        ).scalar_one_or_none()
+
+        # Only create the association if it doesn't already exist
+        if not existing_association:
+            application_user = ApplicationUser(
+                application_id=application.application_id, user_id=user_id
+            )
+            db_session.add(application_user)
+
```

Comment:
Since this endpoint is where we create the application (right above this section), this wouldn't ever find an existing association. We can just add an application user object.

The user_id also can't be null at the moment, so this can be simplified down to something like:
```suggestion
    application_user = ApplicationUser(application=application,user=user)
    db_session.add(application_user)
```